### PR TITLE
Bug/issue 2211

### LINKED ISSUE #2211: start and end timestamps

### PR DESCRIPTION
**Pull request by KolesnikAntony**
close #2211 
Added timestamp to query

### CODE REVIEW

File: frontend/src/lib/getMinMax.ts [hunk 1]
```diff
@@ -6,40 +6,33 @@ import getMinAgo from './getStartAndEndTime/getMinAgo';
 const GetMinMax = (
 	interval: Time,
 	dateTimeRange?: [number, number],
+	// eslint-disable-next-line sonarjs/cognitive-complexity
```
Comment (palashgdev) [inline frontend/src/lib/getMinMax.ts:9]: can you please this disabling of the rule

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/container/TopNav/DateTimeSelection/index.tsx [hunk 2]
```diff
@@ -198,6 +220,63 @@ function DateTimeSelection({
...
+				new Date(parseInt(getTimeString(searchStartTime), 10)),
+			);
+			const { minutedDiff, hoursDiff, daysDiff } = getDiffs(lastRefresh);
+
+			if (daysDiff > 1) {
+				return '1week';
+			}
+
+			if (hoursDiff > 6) {
+				return '1day';
```
Comment (palashgdev) [inline frontend/src/container/TopNav/DateTimeSelection/index.tsx:235]: also i don't think this logic will scale has it is not matching the current available options available for current route

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/hooks/useIntervalRange.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+				return '30min';
+			}
+			if (minutedDiff > 5) {
+				return '15min';
+			}
+			if (minutedDiff > 1) {
+				return '5min';
+			}
+
+			return ROUTES.SERVICE_MAP === pathname ? '1week' : 'custom';
```
Comment (palashgdev) [inline frontend/src/hooks/useIntervalRange.ts:56]: any reason for `ROUTES.SERVICE_MAP` ?

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/hooks/useIntervalRange.ts (hunk 3 above)
Comment (KolesnikAntony) [reply frontend/src/hooks/useIntervalRange.ts:56]: image
I could not see the custom option inside the select on this page

Comment (palashgdev) [reply frontend/src/hooks/useIntervalRange.ts:56]: @KolesnikAntony got it

File: frontend/src/hooks/useIntervalRange.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	const localstorageEndTime = getLocalStorageKey('endTime');
+	const getQueryInterval = useCallback(
+		(searchStartTime: string) => {
+			const lastRefresh = dayjs(
+				new Date(parseInt(getTimeString(searchStartTime), 10)),
+			);
+			const { minutedDiff, hoursDiff, daysDiff } = getDiffs(lastRefresh);
+
+			if (daysDiff > 1) {
+				return '1week';
```
Comment (palashgdev) [inline frontend/src/hooks/useIntervalRange.ts:35]: if dayDiff is greater than 1 then we are returning 1 week don't you think this is scalable let's say in future we add more `Time` for that reason this logic will fail

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/container/TopNav/DateTimeSelection/index.tsx (hunk 2 above)
Comment (palashgdev) [reply frontend/src/container/TopNav/DateTimeSelection/index.tsx]: actually in service map we don't show the custom

**Comment by KolesnikAntony** [comment]
In last commit has been added new logic. First all we get diff between start and current dates. Second step we check max date in our options arrat by type of time [hr, day and etc]. If we have diff more than or option value , we set next param from scope. (For example if we had diff 5day but in our array max it is 1day then we set 1week). If diff is less we get closest bigger value from array.  @palashgdev 
image

image
image
image

File: frontend/.eslintrc.js [hunk 5]
```diff
@@ -58,7 +58,7 @@ module.exports = {
 		'react/no-array-index-key': 'error',
 		'linebreak-style': [
 			'error',
-			process.platform === 'win32' ? 'windows' : 'unix',
+			process?.env.platform === 'win32' ? 'windows' : 'unix',
```
Comment (palashgdev) [inline frontend/.eslintrc.js:61]: can we please raise a another PR for this one

File: frontend/.eslintrc.js [hunk 6]
```diff
@@ -108,7 +108,7 @@ module.exports = {
 		// eslint rules need to remove
 		'@typescript-eslint/no-shadow': 'off',
 		'import/no-cycle': 'off',
-
+		'sonarjs/cognitive-complexity': 'off',
```
Comment (palashgdev) [inline frontend/.eslintrc.js:111]: please remove this one

File: frontend/src/hooks/useIntervalRange.ts [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+			const maxDays = getMaxValue(type);
+			if (count > maxDays) {
+				return alt;
+			}
+			return getOptionValue(count, type);
+		},
+		[getMaxValue, getOptionValue],
+	);
+
+	const getQueryInterval = useCallback(
```
Comment (palashgdev) [inline frontend/src/hooks/useIntervalRange.ts:71]: can we have a test for the same

**Review by palashgdev — CHANGES_REQUESTED** [review]

**Comment by palashgdev** [comment]
> In last commit has been added new logic. First all we get diff between start and current dates. Second step we check max date in our options arrat by type of time [hr, day and etc]. If we have diff more than or option value , we set next param from scope. (For example if we had diff 5day but in our array max it is 1day then we set 1week). If diff is less we get closest bigger value from array. @palashgdev image
> 
> image image
> 
> image image

Thanks @KolesnikAntony this sounds good to me

File: frontend/src/container/TopNav/DateTimeSelection/index.tsx [hunk 8]
```diff
@@ -198,7 +196,10 @@ function DateTimeSelection({
 	};
 
 	// this is triggred when we change the routes and based on that we are changing the default options
+	const { getTime, getCustomOrIntervalTime } = useIntervalRange();
+
 	useEffect(() => {
+		console.log('USE EFFECT');
```
Comment (palashgdev) [inline frontend/src/container/TopNav/DateTimeSelection/index.tsx:202]: please remove this console

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/package.json [hunk 9]
```diff
@@ -31,6 +31,7 @@
 		"@ant-design/icons": "4.8.0",
 		"@grafana/data": "^8.4.3",
 		"@monaco-editor/react": "^4.3.1",
+		"@testing-library/react-hooks": "^8.0.1",
```
Comment (palashgdev) [inline frontend/package.json:34]: exact version

**Comment by palashgdev** [comment]
@KolesnikAntony can you please check build pipeline is failing

File: frontend/src/lib/getMinMax.ts [hunk 10]
```diff
@@ -4,47 +4,50 @@ import { GlobalReducer } from 'types/reducer/globalTime';
...
-	} else {
-		throw new Error('invalid time type');
+	switch (interval) {
+		case 'queried':
+			minTime = (dateTimeRange || [])[1] || 0;
+			break;
+		case '1min':
+			minTime = getMinAgo({ minutes: 1 }).getTime();
+			break;
+		case '15min':
```
Comment (s4nju) [inline frontend/src/lib/getMinMax.ts:20]: Missing 10 mins case here.

**Pull request closed without merging**